packages/jaeger-ui/src/components/DeepDependencyGraph/Header/index.tsx

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as React from 'react';
+
+import NameSelector from './NameSelector';
+import tempOptions from './tmp-data';
+
+import './index.css';
+
+type TProps = {};
+
+type TTempState = {
+  service: string | null;
+  operation: string | null;
+};
+
+// istanbul ignore next
+const TMP_OPTIIONS = tempOptions.map(s => (s.match(/^(\S+\s+){1,12}|^\S+$/) || ['MISSING'])[0]);
+
+export default class Header extends React.PureComponent<TProps, TTempState> {
```

Comment:
I feel as though this PR should include tying the `Service` and `Operation` selectors into the URL state, and the default `start` and `end` values (which can be any two numbers until the BE endpoint takes these values), as before.
Otherwise merging this would break the ability to consume the test data or and block consuming live path data.

Reply:
I think this PR should stand alone, and the functionality you describe here should be in a different PR. Adding that to this PR conflates the PR from being components to including state management.